src/main/java/lotto/handler/InputHandler.java

```diff
@@ -0,0 +1,80 @@
+package lotto.handler;
+
+import java.util.List;
+import lotto.util.InputParser;
+import lotto.util.InputValidator;
+import lotto.model.domain.BonusNumber;
+import lotto.model.domain.PurchaseAmount;
+import lotto.model.domain.WinningNumber;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class InputHandler {
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public InputHandler(InputView inputView, OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    public PurchaseAmount handlePurchaseAmount() {
+        while (true) {
+            try {
+                outputView.requestPurchaseAmount();
+                int purchaseAmount = getPurchaseAmount();
+
+                return new PurchaseAmount(purchaseAmount);
+            } catch (IllegalArgumentException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+    public WinningNumber handleWinningNumber() {
+        while (true) {
+            try {
+                outputView.requestWinningNumber();
+                List<Integer> winningNumber = getWinningNumber();
+
+                return new WinningNumber(winningNumber);
+            } catch (IllegalArgumentException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+    public BonusNumber handleBonusNumber(WinningNumber winningNumber) {
+        while (true) {
+            try {
+                outputView.requestBonusNumber();
+                int bonusNumber = getBonusNumber();
+
+                return new BonusNumber(bonusNumber, winningNumber);
+            } catch (IllegalArgumentException e) {
+                System.out.println(e.getMessage());
+            }
+        }
```

Comment:
이 부분은 
```suggestion
       public <T> T get(Supplier<T> supplier) {
           while (true) {
                  try {
                      return supplier.get()
                  } catch (IllegalArgumentException e) {
                      System.out.println(e.getMessage());
                  }
            }
        }
```
이렇게 바꿀수도 있을거 같습니다

Reply:
3주차 리뷰 남기며 재시도 로직에 함수형 인터페이스를 도입하신 분들이 많더군요! 좋은 피드백 감사합니다:)